### ISSUE TITLE
Add support for Finch in the Kind project

### ISSUE THREAD

**Issue by tsahiduek** [runfinch/finch#361]
**What is the problem you're trying to solve?.**
Kind project cannot be used with finch. It requires running podman and finch altogether 

**Describe the feature you'd like**
Work with the Kind community to add support for finch, the same way it exists for [podman](https://github.com/kubernetes-sigs/kind/tree/main/pkg/cluster/internal/providers/podman)

**Comment by csantanapr** [comment]
@weikequ nice to meet you during Linux OSSummit NA 2023, I'm interested on using kind on macOS using finch also.

Here is the issue I mentioned related to nerdctl https://github.com/containerd/nerdctl/issues/349#issuecomment-1556229062
@BenTheElder made a comment 3 days ago.

Please comment in this issue, if kind support is something the project wants to add and how the community can help with this.
cc @estesp

**Comment by BenTheElder** [comment]
Am I gathering correctly that this tool is used to create a VM running containerd is based on lima, similar to colima, rancher desktop etc?

nerdctl support is something kind is considering, but not committed to yet -- similar VM environments can support docker or podman, and docker in particular has some very helpful additional features that aren't easily replaced (we still don't have full parity with the podman backend), and we require most of our time currently for more general bug fixes / maintenance vs feature work (or feature review).

See the linked threads in nerdctl and from there KIND for more context.

**Comment by estesp** [comment]
> Am I gathering correctly that this tool is used to create a VM running containerd is based on lima, similar to colima, rancher desktop etc?

Yes, Finch = (today) macOS command line client support, using Lima + containerd + nerdctl + BuildKit inside the VM instance (based on Fedora). So, very similar to colima and Rancher, although both those tools can/do provide the Docker engine inside the VM as well/optionally.

> 
> nerdctl support is something kind is considering, but not committed to yet -- similar VM environments can support docker or podman, and docker in particular has some very helpful additional features that aren't easily replaced (we still don't have full parity with the podman backend), and we require most of our time currently for more general bug fixes / maintenance vs feature work (or feature review).
> 
> See the linked threads in nerdctl and from there KIND for more context.

I wonder if this table (https://github.com/containerd/nerdctl/issues/349#issuecomment-979015373) is still the right list of missing features or if anything has changed in 18 months? Nerdctl has had a lot of development since then and is at version v1.4.0 (that table notes the version at the time being v0.14.0) // cc: @AkihiroSuda

**Comment by Shubhranshu153** [comment]
The pr has been merged in nerdctl.
